### PR TITLE
Show a key error if a dict error happens

### PR DESCRIPTION
**Pull request by her0e1c1**
I added a key error because if a dict is huge, it's hard to find what key error happen

### CODE REVIEW

File: schema.py [hunk 1]
```diff
@@ -237,10 +237,15 @@ def validate(self, data):
...
                         pass
                     else:
-                        nvalue = Schema(svalue, error=e).validate(value)
-                        new[nkey] = nvalue
-                        coverage.add(skey)
-                        break
+                        try:
+                            nvalue = Schema(svalue, error=e).validate(value)
+                        except SchemaError as x:
+                            k = "Key %s error:" % nkey
```
Comment (skorokithakis) [inline schema.py]: Can you add quotes around `%s`, to improve readability?

File: test_schema.py [hunk 2]
```diff
@@ -240,6 +240,21 @@ def test_dict_subtypes():
     # is dropped!
 
 
+def test_dict_key_error():
```
Comment (skorokithakis) [inline test_schema.py]: Looks like tests are failing, for some reason.

File: schema.py (hunk 1 above)
Comment (her0e1c1) [inline schema.py]: @skorokithakis thanks for your feedback 😄  I added quotes

File: test_schema.py (hunk 2 above)
Comment (her0e1c1) [inline test_schema.py]: I fixed the doctest error so I hope it works this time!

File: test_schema.py [hunk 3]
```diff
@@ -240,6 +240,22 @@ def test_dict_subtypes():
...
+    try:
+        Schema({'k': int}).validate({'k': 'x'})
+    except SchemaError as e:
+        assert e.code == "Key 'k' error:\n'x' should be instance of 'int'"
+    try:
+        Schema({'k': {'k2': int}}).validate({'k': {'k2': 'x'}})
+    except SchemaError as e:
+        code = "Key 'k' error:\nKey 'k2' error:\n'x' should be instance of 'int'"
```
Comment (skorokithakis) [inline test_schema.py]: Can you run pep8 on these files? It complains that some lines are too long :(

Comment (her0e1c1) [inline test_schema.py]: Yes. It was longer than 90 before, so I stored the error message to `code` variable.
So I think my latest commit resolves PEP8

Comment (her0e1c1) [inline test_schema.py]: @skorokithakis I think this PEP8 looks ok.
https://travis-ci.org/keleshev/schema/jobs/183802776
So please tell me a problem if it still exists. I want to fix it

**Comment by skorokithakis** [comment]
I'm sorry, yes, this looks good! A test is failing, but it looks like it's because of Travis.

Thanks for your help!

**Comment by her0e1c1** [comment]
Thanks for your merge 😄  Our product becomes better now!!!

**Pull request merged**